bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+        },
+        {
+            name:"denyTimeOff",
+            parameters:[
+                {name:"comment", type:"STRING"},
+                {name:"personNumber", type:"STRING"},
+                {name:"userText", type:"STRING"},
+                {name:"requestId", type:"STRING"},
+            ],
+            function : denyTimeOff
+        },
+        {
+            name:"recordtimestamp",
+            parameters:[
+                {name:"personNumber", type:"STRING"},
+                {name:"time", type:"STRING"},
+                {name:"date", type:"STRING"}
+            ],
+            function : recordTimestamp
+        },
+        {
+            name:"approveTimeLog",
+            parameters:[
+                {name:"comment", type:"STRING"},
+                {name:"personNumber", type:"STRING"},
+                {name:"userText", type:"STRING"},
+                {name:"requestId", type:"STRING"},
+            ],
+            function : approveTimelog
+        },
+        {
+            name:"sentBackTimeLog",
+            parameters:[
+                {name:"comment", type:"STRING"},
+                {name:"personNumber", type:"STRING"},
+                {name:"userText", type:"STRING"},
+                {name:"requestId", type:"STRING"},
+            ],
+            function : sentBackTimelog
+        },
+    ]
+});
+
+async function updateAccrualBalance({ actionParameters, client, dataStore,integrationParameters}) {
+    integrationParameters.dateFormat = getValidDateFormat(integrationParameters.dateFormat)
+    const date = actionParameters.date;
+    const number = actionParameters.personNumber;
+    let accrualBalance = [];
+    const session = await logIn(client, integrationParameters.username , integrationParameters.password);
+    const request = createRequest(client,session);
+    const body = `<?xml version='1.0' encoding='UTF-8' ?>
+    <Kronos_WFC version="1.0">
+        <Request Action="Load">
+            <AccrualData>
+                <BalanceDate>${moment(date).format(integrationParameters.dateFormat)}</BalanceDate>
+                <Employee>
+                    <PersonIdentity personNumber="${number}"/>
+                </Employee>
+            </AccrualData>
+        </Request>
+    </Kronos_WFC>`
+    const {response,result} = await request(body)
+    let dt = result.Kronos_WFC.Response[0].AccrualData[0].AccrualBalances[0].AccrualBalanceSummary ? result.Kronos_WFC.Response[0].AccrualData[0].AccrualBalances[0].AccrualBalanceSummary : [];
+    dt.forEach(k => {
+        accrualBalance.push({
+        "accrualCodeId": k.$.AccrualCodeId,
+        "accrualCodeName": k.$.AccrualCodeName,
+        "accrualType": k.$.AccrualType,
+        "encumberedBalanceInTime": Number(k.$.EncumberedBalanceInTime),
+        "hoursPerDay": k.$.HoursPerDay,
+        "projectedVestedBalanceInTime": k.$.ProjectedVestedBalanceInTime,
+        "projectedDate": moment(actionParameters.date).format('yyyy-MM-DD'),
```

Comment:
Is it ok that the date format is hardcoded here?
Shouldn't it be same as in `syncAccrualBalance()`? -> `integrationParameters.dateFormat`?